client/pages/clubs/[id].tsx

```diff
@@ -1,22 +1,251 @@
+  display: flex;
+`
+
+const Category = styled.p`
+  border: 1px solid black;
+  border-radius: 15px;
+  font-size: 12px;
+  margin-left: 5px;
+
+  padding: 5px 10px;
+  @media ${device.mobileS} {
+    padding: 2px 5px;
+  }
+`
+
+const ClubDescription = styled.p`
+  margin-top: 24px;
+`
+
+interface ClubInfoProps {
+  club: Club
 }
+
+const ClubInfo = ({ club }: ClubInfoProps) => {
+  const { name, description, iconURL, tags } = club
+  const links = [club.facebookLink, club.twitterLink, club.instagramLink, club.websiteLink]
+  const linksThatExist = links.filter((l) => l !== undefined)
+
+  return (
+    <div>
+      <BackArrow src="/back-arrow.svg" width="28px" />
+
+      <Logo src={iconURL} alt="" />
+      <ClubName> {name}</ClubName>
+
+      <Links>
+        {linksThatExist.map((l) => (
+          <SocialLink>{l}</SocialLink>
+        ))}
+      </Links>
+
+      <ClubMetaData>
+        <MemberCount>20-30 members</MemberCount>
+        <Categories tags={tags} />
+        {/* TODO: color code the tags */}
+      </ClubMetaData>
+
+      <ClubDescription>{description}</ClubDescription>
+    </div>
+  )
+}
+
+const Categories = ({ tags }: { tags: Array<string> }) => (
+  <CategoriesWrapper>
+    {tags.map((t) => (
+      <Category>{t}</Category>
+    ))}
+  </CategoriesWrapper>
+)
+
+const EventsTitle = styled.h2`
+  margin-top: 32px;
+`
+
+const EventCardDetails = styled.div`
+  margin: 0 20px 20px;
+  width: 100%;
+
+  @media ${device.mobileS} {
+    margin: 0 10px 10px;
+  }
+`
+
+const EventCardTitle = styled.p`
+  font-weight: ${weight700};
+  margin-bottom: 0;
+`
+
+// TODO: make box shadow cleaner
+const EventCardWrapper = styled.div`
+  box-shadow: 0 5px 10px rgba(154, 160, 185, 0.05), 0 15px 100px rgba(166, 173, 201, 0.2);
+  display: flex;
+  margin-top: 10px;
+`
+
+const EventCardImg = styled.img`
+  width: 256px;
+  @media ${device.mobileL} {
+    width: 128px;
+  }
+`
+
+const EventCardDate = styled.p`
+  margin-top: 5px;
+`
+
+const PastEventsTitle = styled.p`
+  font-size: ${desktopFontSize.subtitle1};
+`
+
+const EventsHostedByClub = ({ events }: { events: Array<BasicEvent> }) => {
+  const upcomingEvents = events.filter((e) => e.startTime.isAfter(moment()))
+
+  const pastEvents = events.filter((e) => e.startTime.isBefore(moment()))
+
+  return (
+    <div>
+      <EventsTitle>Upcoming Events</EventsTitle>
+      {upcomingEvents.map((e) => {
+        return <EventCard event={e} />
+      })}
+      <PastEventsTitle>Past Events</PastEventsTitle>
+      {pastEvents.map((e) => {
+        return <EventCard event={e} />
+      })}
+    </div>
+  )
+}
+
+const EventCard = ({ event }: { event: BasicEvent }) => (
+  <EventCardWrapper>
+    <EventCardImg src={event.backgroundImageURL} alt="" />
+    <EventCardDetails>
+      <EventCardTitle>{event.name}</EventCardTitle>
+      <EventCardDate>
+        {device.mobileS ? (
+          <p>
+            {event.startTime.format('dddd')}, {event.startTime.format('LL')}
```

Comment:
`{event.startTime.format('dddd')}, {event.startTime.format('LL')}` this looks to be a constant that's used in both the mobile and large-screen case. Maybe we can factor this out into a string constant outside the JSX?

Likewise with `{event.startTime.format('LT')} - {event.endTime.format('LT')}`

Reply:
Done